### PR TITLE
Create import graph of types and namespaces

### PR DESCRIPTION
**Pull request by delvedor**
This pr adds a new compiler step that generates two additional JSONs, one with the import graph of every type and another one with the import graph of every namespace.
From this data, we will generate a UI to visualize how different modules interacts with each other.

TODO:
- [x] generate a raw data JSON report
- [x] generate a graphical report to visualize dependencies
- [x] publish UI report to github pages

Closes: #339

### CODE REVIEW

File: compiler/src/steps/import-graph.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        default:
+      }
+    }
+  }
+
+  const typeGraph: ImportTypesGraph[] = []
+  for (const [type, types] of importGraph.entries()) {
+    typeGraph.push({
+      type,
+      imported_by: types.map(type => type.name)
```
Comment (delvedor) [inline compiler/src/steps/import-graph.ts:97]: TODO: this should be a unique array.

File: compiler/src/steps/import-graph.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+interface ImportNamespaceGraph {
+  namespace: string
+  imported_by: string[]
+}
+
+export default async function createImportGraph (model: model.Model, jsonSpec: Map<string, JsonSpec>): Promise<model.Model> {
+  const importGraph: Map<model.TypeName, model.TypeDefinition[]> = new Map()
+  for (const current of model.types) {
```
Comment (swallez) [inline compiler/src/steps/import-graph.ts:37]: It's more complex than that, as we have to traverse every `ValueOf` (extends, generics, property types, etc) to build the graph from `TypeName` values.

File: output/schema/import-namespace-graph.json [hunk 3]
```diff
@@ -0,0 +1,10954 @@
+[
+  {
+    "namespace": "_global.bulk",
```
Comment (philkra) [inline output/schema/import-namespace-graph.json]: https://github.com/elastic/elasticsearch-specification/pull/1261#discussion_r784709089 assume the same holds true for the namespace. Should be unique

**Review by philkra — CHANGES_REQUESTED** [review]

**Comment by delvedor** [comment]
The import map is not committed in the repo for now, it adds a lot of data that is only used for generating the graph (which is published on GH pages).
We'll generate some markdown representations of the json that will be published on GH pages as well.

**Review by swallez — APPROVED** [review]
LGTM!

**Review by philkra — APPROVED** [review]
LGTM

**Pull request merged**